README.md
# @polyscale/polyscale-node

Client for interacting with the PolyScale API.

## Example Usage

```typescript
import PolyScale from "@polyscale/polyscale-node";

const polyscale = new PolyScale({
  url: "https://url-to-api.com",
  apiKey: "some-api-key",
});

/**
 * Basic usage
 */
polyscale.cache
  .create({
    name: "Example Cache",
    host: "localhost",
    port: 4000,
    workspaceId: "some-id",
  })
  .then((cache) => {
    console.dir(cache);

    polyscale.cacheTtl.create(cache.id, {
      type: "TABLE",
      value: 5,
      key: "tableA",
    });
  })
  .catch((error) => {
    console.error(error);
  });

/**
 * Using Async/Await
 */
const init = async () => {
  try {
    const cache = await polyscale.cache.create({
      name: "Example Cache",
      host: "localhost",
      port: 4000,
      workspaceId: "some-id",
    });

    console.dir(cache);

    await polyscale.cacheTtl.create(cache.id, {
      type: "TABLE",
      value: 5,
      key: "tableA",
    });
  } catch (error) {
    console.error(error);
  }
};

init();
```

## License

polyscale-node is [MIT licensed](./LICENSE).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.0.23](https://github.com/polyscale/polyscale-node/compare/v0.0.22...v0.0.23) (2023-10-04)


### Features

* bump node-fetch ([66be179](https://github.com/polyscale/polyscale-node/commit/66be17938826c4699e0ca1a386a3dea9f99c854b))

### [0.0.22](https://github.com/polyscale/polyscale-node/compare/v0.0.21...v0.0.22) (2023-06-02)


### Bug Fixes

* add build files ([fcab633](https://github.com/polyscale/polyscale-node/commit/fcab63355fd226f466b0eb334ace038fdb7f9d15))

### [0.0.21](https://github.com/polyscale/polyscale-node/compare/v0.0.20...v0.0.21) (2023-06-02)


### Features

* add new property type ([167742e](https://github.com/polyscale/polyscale-node/commit/167742ee208b9de838bc918837bce8248827cf5a))
* publish to npm ([7353b20](https://github.com/polyscale/polyscale-node/commit/7353b206a8f737d47543db9da9d979670448a2f2))

### [0.0.20](https://github.com/polyscale/polyscale-node/compare/v0.0.19...v0.0.20) (2022-09-14)

### [0.0.19](https://github.com/polyscale/polyscale-node/compare/v0.0.18...v0.0.19) (2022-09-13)

### [0.0.18](https://github.com/polyscale/polyscale-node/compare/v0.0.17...v0.0.18) (2022-09-13)

### [0.0.17](https://github.com/polyscale/polyscale-node/compare/v0.0.16...v0.0.17) (2022-09-13)


### Features

* mssql graphql caches ([ef2ea3f](https://github.com/polyscale/polyscale-node/commit/ef2ea3fe20003d6fbbc82656b3acf5fd00b2fc52))

### [0.0.16](https://github.com/polyscale/polyscale-node/compare/v0.0.15...v0.0.16) (2022-02-10)


### Features

* make example testable ([f623f1f](https://github.com/polyscale/polyscale-node/commit/f623f1fecbcdb99367a194af5d165ff31f752ba5))

### [0.0.15](https://github.com/polyscale/polyscale-node/compare/v0.0.14...v0.0.15) (2022-02-09)


### Bug Fixes

* support auto/manual ttl ([6358bc8](https://github.com/polyscale/polyscale-node/commit/6358bc8d6ff24661e4594777f6f510736405e18e))

### [0.0.14](https://github.com/polyscale/polyscale-node/compare/v0.0.13...v0.0.14) (2022-02-09)


### Features

* defaultCachingBehaviour ([648b6c0](https://github.com/polyscale/polyscale-node/commit/648b6c04cb87df2f19eea0e7f81f724eb5dd8edd))

### [0.0.13](https://github.com/polyscale/polyscale-node/compare/v0.0.12...v0.0.13) (2022-01-25)


### Features

* support database/cachingEnabled/autoEnabled ([558a423](https://github.com/polyscale/polyscale-node/commit/558a423d494f823914061bd6c624641158ab0df3))

### [0.0.12](https://github.com/polyscale/polyscale-node/compare/v0.0.11...v0.0.12) (2021-10-20)


### Bug Fixes

* empty purge body ([6f5ece5](https://github.com/polyscale/polyscale-node/commit/6f5ece59eaa56f479108c38c5eea5a7a8430c08c))

### [0.0.11](https://github.com/polyscale/polyscale-node/compare/v0.0.10...v0.0.11) (2021-10-15)


### Features

* purge cache ([f0dc6e5](https://github.com/polyscale/polyscale-node/commit/f0dc6e5a6bbf0a4d734d4f2fe3dea78a96f7a62e))

### [0.0.10](https://github.com/polyscale/polyscale-node/compare/v0.0.9...v0.0.10) (2021-10-15)

### [0.0.9](https://github.com/polyscale/polyscale-node/compare/v0.0.8...v0.0.9) (2021-08-03)


### Bug Fixes

* removed postinstall script ([d0b70eb](https://github.com/polyscale/polyscale-node/commit/d0b70eb84f69a7d134b057dbacd65863cbb0856d))

### [0.0.8](https://github.com/polyscale/polyscale-node/compare/v0.0.7...v0.0.8) (2021-08-03)


### Bug Fixes

* drop id from create cache ([1367de2](https://github.com/polyscale/polyscale-node/commit/1367de29b3a66905b4e08bcab215a5de2a302a15))
* drop workspace id ([4f98a63](https://github.com/polyscale/polyscale-node/commit/4f98a63f01dd3558984c4868f21a371193cdddbf))

### [0.0.7](https://github.com/polyscale/polyscale-node/compare/v0.0.6...v0.0.7) (2021-08-02)


### Bug Fixes

* build for js ([f301164](https://github.com/polyscale/polyscale-node/commit/f3011648cd58a6145452a58f4d706df2c73ebab5))
* husky precommit hook ([1e4d2d5](https://github.com/polyscale/polyscale-node/commit/1e4d2d5dad0a11c28fb85b20bec1d32a802d65df))

### [0.0.6](https://github.com/polyscale/polyscale-node/compare/v0.0.5...v0.0.6) (2021-07-30)


### Bug Fixes

* correctly forward api errors ([7baf630](https://github.com/polyscale/polyscale-node/commit/7baf6300b183235b90aefbb1dcfbd92ee30219bc))

### [0.0.5](https://github.com/polyscale/polyscale-node/compare/v0.0.4...v0.0.5) (2021-07-30)


### Features

* more concise function signatures ([ef94c5f](https://github.com/polyscale/polyscale-node/commit/ef94c5fc2618d50fd046d119e272ba2d808568e4))


### Bug Fixes

* add build files ([09e9e26](https://github.com/polyscale/polyscale-node/commit/09e9e26b97defca0c3dcd7bd1e02854a0b4370c4))

### [0.0.4](https://github.com/polyscale/polyscale-node/compare/v0.0.3...v0.0.4) (2021-07-29)


### Features

* new cache key format ([dce5ac2](https://github.com/polyscale/polyscale-node/commit/dce5ac266372eac8f9f2052818d863ede032e4c8))

### [0.0.3](https://github.com/polyscale/polyscale-node/compare/v0.0.2...v0.0.3) (2021-07-29)


### Features

* optional id on cache creation ([f8804af](https://github.com/polyscale/polyscale-node/commit/f8804afbf1182600d2312e69aef16066d8edf60f))

### [0.0.2](https://github.com/polyscale/polyscale-node/compare/v0.0.1...v0.0.2) (2021-07-28)


### Bug Fixes

* update main ([58f1a7c](https://github.com/polyscale/polyscale-node/commit/58f1a7c2a19985c702414db976ed36c3b53006b1))

### 0.0.1 (2021-07-26)

### [0.0.1](https://github.com/polyscale/polyscale-node/compare/v0.0.0...v0.0.1) (2021-07-22)


### Features

* restructure module ([9d9d8cf](https://github.com/polyscale/polyscale-node/commit/9d9d8cfbc1a8d6006799067e8b82e0eeb765f6ca))


### Bug Fixes

* husky missing ([78a5268](https://github.com/polyscale/polyscale-node/commit/78a5268c534bec2bb7c5ad51019aa7b78cb86188))

example/example.ts
import assert from "assert";
import { CacheTtl, PolyScale } from "../src/main";

const polyscale = new PolyScale({
  url: "http://localhost:8000",
  apiKey: process.env.API_KEY ?? "some-api-key",
});

/**
 * Using Async/Await
 */
const init = async () => {
  try {
    const cache = await polyscale.cache.create({
      name: "Example Cache",
      host: "localhost",
      port: 4000,
      database: "mysql",
      cachingEnabled: true,
      defaultCachingBehaviour: "autoCached",
    });

    console.dir(cache);

    let ttls: Array<CacheTtl>;

    await polyscale.cacheTtl.create({
      cacheId: cache.id,
      type: "AUTO",
      key: "some-template",
    });

    ttls = await polyscale.cacheTtl.getMany({ cacheId: cache.id });

    assert(ttls.length === 1, "Expected 1 ttl");
    assert(
      ttls.find((d) => d.type === "AUTO"),
      "Expected AUTO ttl"
    );

    await polyscale.cacheTtl.create({
      cacheId: cache.id,
      type: "MANUAL",
      key: "some-template",
    });

    ttls = await polyscale.cacheTtl.getMany({ cacheId: cache.id });

    assert(ttls.length === 1, "Expected 1 ttl");
    assert(
      ttls.find((d) => d.type === "MANUAL"),
      "Expected MANUAL ttl"
    );

    await polyscale.cacheTtl.create({
      cacheId: cache.id,
      type: "TABLE",
      value: 5,
      key: "tableA",
    });

    ttls = await polyscale.cacheTtl.getMany({ cacheId: cache.id });

    assert(ttls.length === 2, "Expected 2 ttls");

    await polyscale.cacheTtl.update({
      cacheId: cache.id,
      cacheTtlKey: "tableA",
      update: {
        value: 30,
      },
    });

    await polyscale.cacheTtl.delete({
      cacheId: cache.id,
      cacheTtlKey: "tableA",
      cacheTtlType: "TABLE",
    });

    await polyscale.cacheTtl.delete({
      cacheId: cache.id,
      cacheTtlKey: "some-template",
      cacheTtlType: "MANUAL",
    });

    ttls = await polyscale.cacheTtl.getMany({ cacheId: cache.id });

    assert(ttls.length === 0, "Expected 0 ttls");

    await polyscale.cache.delete({
      cacheId: cache.id,
    });
  } catch (error) {
    console.error(error);
  }
};

void init();
